### PR TITLE
Basic unit test to go through all the cobra related code

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: cmd/skaffold/app/skaffold_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	defer func(args []string) { os.Args = args }(os.Args)
+	os.Args = []string{"skaffold", "help"}
+
+	err := Run(&output, &errOutput)
+
+	testutil.CheckError(t, false, err)
+	testutil.CheckContains(t, "Available Commands", output.String())
+	testutil.CheckDeepEqual(t, "", errOutput.String())
+}
```
Comment (tejal29) [inline cmd/skaffold/app/skaffold_test.go]: Can we also have a test case for invalid command and see if error is returned.

Comment (dgageot) [reply cmd/skaffold/app/skaffold_test.go]: Added!

Comment (dgageot) [reply cmd/skaffold/app/skaffold_test.go]: @tejal29 woot! Discovered a race condition thanks to this test

File: cmd/skaffold/app/cmd/cmd.go [hunk 2]
```diff
@@ -38,24 +38,28 @@ var (
...
+	}
+
 	rootCmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
 		if err := SetUpLogs(err, v); err != nil {
 			return err
 		}
 		rootCmd.SilenceUsage = true
 		logrus.Infof("Skaffold %+v", version.Get())
+
+		// Avoid race condition
```
Comment (tejal29) [inline cmd/skaffold/app/cmd/cmd.go]: Can you describe what race condition is it avoiding for dummies like me?

Comment (dgageot) [reply cmd/skaffold/app/cmd/cmd.go]: I'm going to simplify the code so that this is not needed anymore.

**Pull request merged**